dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -582,10 +687,16 @@ DecodeDetail ExchangeReceiverBase<RPCContext>::decodeChunks(
 }
 
 template <typename RPCContext>
-ExchangeReceiverResult ExchangeReceiverBase<RPCContext>::nextResult(std::queue<Block> & block_queue, const Block & header)
+ExchangeReceiverResult ExchangeReceiverBase<RPCContext>::nextResult(std::queue<Block> & block_queue, const Block & header, size_t stream_id)
 {
+    if (unlikely(stream_id >= msg_channels.size()))
+    {
+        LoggerPtr log = Logger::get("ExchangeReceiver", exc_log->identifier());
+        LOG_FMT_ERROR(log, "stream_id out of range, stream_id: {}, total_stream_count: {}", stream_id, msg_channels.size());
```

Comment:
can we check it in `DAGQueryBlockInterpreter::handleExchangeReceiver`?

Reply:
This is just a guard for the following code.